app/controllers/reservations_controller.rb

```diff
@@ -183,7 +183,24 @@ def update
       return
     end
 
-    @reservation.assign_times_from_params(params[:reservation])
+    reservation_params = params[:reservation]
+    if !@reservation.reserve_start_at_editable?
+      reservation_params.tap do |p|
+        p[:reserve_start_date] = @reservation.reserve_start_date
+        p[:reserve_start_hour] = @reservation.reserve_start_hour
+        p[:reserve_start_min] = @reservation.reserve_start_min
+        p[:reserve_start_meridian] = @reservation.reserve_start_meridian
+      end
+    end
+
+    @reservation.assign_times_from_params(reservation_params)
+
+    validator = Reservations::DurationChangeValidations.new(@reservation)
+    if validator.invalid?
```

Comment:
On thing that might be nice is to make a `render_edit` method that does `set_window` and `render :edit`. Then you could use one method here and at the bottom of `update`.

So this would become something like
```ruby
render_update and return unless duration_change_valid?
```
then
```ruby
def duration_change_valid?
   Reservations::DurationChangeValidations.new(@reservation).valid?
end
```

It might be worth reworking `invalid_for_update?` to be a positive. Not a big deal, I generally prefer positive method names than negative. Unless that leads you to try something like `unless/else`. That way lies the path to madness.